src/PostgREST/Query/QueryBuilder.hs

```diff
@@ -151,13 +151,25 @@ mutateRequestToQuery (Update mainQi uCols body logicForest pkFlts range ordts re
     rangeCols = BS.intercalate ", " ((\col -> pgFmtIdent col <> " = (SELECT " <> pgFmtIdent col <> " FROM pgrst_update_body) ") <$> S.toList uCols)
     (whereRangeIdF, rangeIdF) = mutRangeF mainQi (fst . otTerm <$> ordts)
 
-mutateRequestToQuery (Delete mainQi logicForest range ordts returnings)
+mutateRequestToQuery (Delete mainQi body logicForest pkFlts range ordts returnings)
+  -- The body is not mandatory, although a delete without filters is not possible.
   | range == allRange =
+    let whereLogic | null logicForest = if hasEmptyBody || null pkFlts then "FALSE" else pgrstDeleteBodyF
+                   | otherwise        = (if hasEmptyBody || null pkFlts then mempty else pgrstDeleteBodyF <> "AND") <> logicForestF in
```

Comment:
```suggestion
                   | otherwise        = (if hasEmptyBody || null pkFlts then mempty else pgrstDeleteBodyF <> " AND ") <> logicForestF in
```